src/raiden_libs/register_service.py

```diff
@@ -65,48 +82,86 @@ def main(
 
     # Register if not yet done
     if current_deposit <= 0:
-        log.info("Address not registered in ServiceRegistry")
-
-        deposit_token_address = contracts[CONTRACT_USER_DEPOSIT].functions.token().call()
-        deposit_token_contract = web3.eth.contract(
-            address=deposit_token_address,
-            abi=CONTRACT_MANAGER.get_contract_abi(CONTRACT_CUSTOM_TOKEN),
+        deposit_to_registry(
+            web3=web3,
+            service_registry_contract=service_registry_contract,
+            user_deposit_contract=contracts[CONTRACT_USER_DEPOSIT],
+            service_address=service_address,
         )
 
-        # Check current token balance
-        account_balance = deposit_token_contract.functions.balanceOf(service_address).call()
-        log.info("Current account balance", balance=account_balance, desired_deposit=deposit)
+    update_service_url(
+        web3=web3,
+        service_registry_contract=service_registry_contract,
+        service_address=service_address,
+        service_url=service_url,
+        current_url=current_url,
+    )
+
+    current_deposit = service_registry_contract.functions.deposits(service_address).call()
+    current_url = service_registry_contract.functions.urls(service_address).call()
 
-        # mint tokens if necessary
-        if account_balance < deposit:
-            checked_transact(
-                web3=web3,
-                service_address=service_address,
-                function_call=deposit_token_contract.functions.mint(deposit),
-                task_name="Minting new Test RDN tokens",
-            )
+    log.info("Updated infos", current_deposit=current_deposit, current_url=current_url)
 
-            account_balance = deposit_token_contract.functions.balanceOf(service_address).call()
-            log.info("Updated account balance", balance=account_balance, desired_deposit=deposit)
 
-        # Approve token transfer
+def deposit_to_registry(
+    web3: Web3,
+    service_registry_contract: Contract,
+    user_deposit_contract: Contract,
+    service_address: Address,
+) -> None:
+    log.info("Address not registered in ServiceRegistry")
```

Comment:
how do you know that? Do you check for duplicates somewhere?

Reply:
`deposit_to_registry` is only called if the check you commented on above is false.